From d939d14218283a4f42ca9d070c486b6ef84e4ea8 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 8 Sep 2014 13:08:46 -0400
Subject: [tools] perf/report: Add mem-mode documentation to report command

Message-id: <1410181752-25631-282-git-send-email-jolsa@redhat.com>
Patchwork-id: 91887
O-Subject: [PATCH RHEL7.1 BZ1134356 281/307] perf report: Add mem-mode documentation to report command
Bugzilla: 1134356
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Don Zickus <dzickus@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1134356
https://bugzilla.redhat.com/show_bug.cgi?id=1134356

upstream
========
commit 75e906c9601aee73b88d6e6dc02371f8c3ca24d7
Author: Don Zickus <dzickus@redhat.com>
Date: Fri May 23 18:41:23 2014 +0200

description
===========
Add mem-mode sorting types and mem-mode itself to perf-report documentation.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/Documentation/perf-report.txt b/tools/perf/Documentation/perf-report.txt
index cefdf43..00fbfb6 100644
--- a/tools/perf/Documentation/perf-report.txt
+++ b/tools/perf/Documentation/perf-report.txt
@@ -117,6 +117,21 @@ OPTIONS
  By default, every sort keys not specified in -F will be appended
  automatically.
 
+ If --mem-mode option is used, following sort keys are also available
+ (incompatible with --branch-stack):
+ symbol_daddr, dso_daddr, locked, tlb, mem, snoop.
+
+ - symbol_daddr: name of data symbol being executed on at the time of sample
+ - dso_daddr: name of library or module containing the data being executed
+ on at the time of sample
+ - locked: whether the bus was locked at the time of sample
+ - tlb: type of tlb access for the data at the time of sample
+ - mem: type of memory access for the data at the time of sample
+ - snoop: type of snoop (if any) for the data at the time of sample
+
+ And default sort keys are changed to local_weight, mem, sym, dso,
+ symbol_daddr, dso_daddr, snoop, tlb, locked, see '--mem-mode'.
+
 -p::
 --parent=<regex>::
         A regex filter to identify parent. The parent is a caller of this
@@ -260,6 +275,13 @@ OPTIONS
  Demangle symbol names to human readable form. It's enabled by default,
  disable with --no-demangle.
 
+--mem-mode::
+ Use the data addresses of samples in addition to instruction addresses
+ to build the histograms.  To generate meaningful output, the perf.data
+ file must have been obtained using perf record -d -W and using a
+ special event -e cpu/mem-loads/ or -e cpu/mem-stores/. See
+ 'perf mem' for simpler access.
+
 --percent-limit::
  Do not show entries which have an overhead under that percent.
  (Default: 0).
-- 
1.7.1